tokenmagic/module/tokenmagic.js

```diff
@@ -608,12 +640,45 @@ export function TokenMagic() {
         placeable._TMFXsetRawFilters(filter);
     };
 
-    function _assignFilters(placeable, filters) {
+    function _loadPersistedValues(params, animeInfos) {
+
+        if (!params.hasOwnProperty("animated")) { return; }
+        if (!animeInfos || animeInfos.length <= 0) { return; }
+
+        for (const effect of Object.keys(params.animated)) {
+
+            for (const animeInfo of animeInfos.values()) {
+
+                if (animeInfo.tmFilterId === params.filterId
+                    && animeInfo.tmFilterInternalId === params.filterInternalId
+                    && animeInfo.tmFilterEffect === effect) {
+                    params.animated[effect].active = false;
+                    params[effect] = animeInfo.tmFilterEffectValue;
+
+                    // special case for halfCosOscillation
+                    if (params.animated[effect].animType === "halfCosOscillation") {
+                        if (params.animated[effect].val1 !== animeInfo.tmFilterEffectValue) {
+                            params.animated[effect].val2 = params.animated[effect].val1;
+                            params.animated[effect].val1 = animeInfo.tmFilterEffectValue;
+                        }
+                    }
+                }
+            }
+        }
+    }
+

```

Comment:
I can't persist all the animation values, it's forbidden :)
But when an effect is finished (when its number of loops is 0), I store the value of the animated property and reload it if necessary (scene change, reload, enter a world), to prevent the effect to trigger again, and to give the property its final state.